shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -76,219 +72,256 @@
   private FrameLayout container;
   private FlutterView flutterView;
-  private PlatformPlugin platformPlugin;
-  private BasicMessageChannel<String> mFlutterLifecycleChannel;
-  private BasicMessageChannel<Object> mFlutterSystemChannel;
-  private MethodChannel mFlutterNavigationChannel;
   private View launchView;
 
-  @Override
-  public View onCreateView(LayoutInflater inflater, ViewGroup parent, Bundle savedInstanceState) {
-    Log.e(TAG, "onCreateView()");
-    createLayout();
-    // TODO: Should we start running the FlutterView here or when attached to window? It was being done here
-    //       in the Activity, but maybe that was a problem to begin with?
-    flutterView.attachToFlutterRenderer(flutterEngine.getRenderer(), flutterEngine.getDartExecutor());
-    doInitialFlutterViewRun();
+  private PlatformPlugin platformPlugin;
 
-    return container;
+  public FlutterFragment() {
+    // Ensure that we at least have an empty Bundle of arguments so that we don't
+    // need to continually check for null arguments before grabbing one.
+    setArguments(new Bundle());
   }
 
   @Override
-  public void onStart() {
-    super.onStart();
-    Log.d(TAG, "onStart()");
-    mFlutterLifecycleChannel.send("AppLifecycleState.inactive");
-  }
+  public void onAttach(Activity activity) {
+    super.onAttach(activity);
 
-  public void onPostResume() {
-    Log.d(TAG, "onPostResume()");
-    flutterView.updateAccessibilityFeatures();
-    platformPlugin.onPostResume();
-    mFlutterLifecycleChannel.send("AppLifecycleState.resumed");
-  }
+    // TODO(mattcarroll): I think the build system is linking the wrong Fragment API. It says that
+    //                    getContext() cannot be found...
+    initializeFlutter(activity);
 
-  @Override
-  public void onPause() {
-    super.onPause();
-    Log.d(TAG, "onPause()");
-    mFlutterLifecycleChannel.send("AppLifecycleState.inactive");
-  }
+    // When "retain instance" is true, the FlutterEngine will survive configuration
+    // changes. Therefore, we create a new one only if one does not already exist.
+    if (flutterEngine == null) {
+      createFlutterEngine();
+    }
 
-  @Override
-  public void onStop() {
-    super.onStop();
-    Log.d(TAG, "onStop()");
-    mFlutterLifecycleChannel.send("AppLifecycleState.paused");
+    // Regardless of whether or not a FlutterEngine already existed, the PlatformPlugin
+    // is bound to a specific Activity. Therefore, it needs to be created and configured
+    // every time this Fragment attaches to a new Activity.
+    connectFlutterEngineToAndroidPlatform();
   }
 
-  @Override
-  public void onDestroy() {
-    super.onDestroy();
-    Log.d(TAG, "onDestroy()");
+  // TODO(mattcarroll): does it really make sense to run this for every Fragment? Note: the
+  //                    implementation of the "ensure" method includes a reference to the default
+  //                    app bundle path. How does that jive with "doInitialFlutterViewRun"?
```

Comment:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-checking-in-comments-that-ask-questions